pkg/controller/common/rollout/workloads/cloneset_controller.go

```diff
@@ -148,37 +161,64 @@ func (c *CloneSetController) CheckOneBatchPods(ctx context.Context) *v1alpha1.Ro
 	if currentBatch.MaxUnavailable != nil {
 		unavail, _ = intstr.GetValueFromIntOrPercent(currentBatch.MaxUnavailable, int(cloneSetSize), true)
 	}
-	klog.InfoS("checking the rolling out progress", "new pod count target", newPodTarget,
-		"new ready pod count", readyPodCount, "max unavailable pod allowed", unavail)
+	klog.V(common.LogDebug).InfoS("checking the rolling out progress", "current batch", currentBatch,
+		"new pod count target", newPodTarget, "new ready pod count", readyPodCount,
+		"max unavailable pod allowed", unavail)
 	c.rolloutStatus.UpgradedReadyReplicas = int32(readyPodCount)
 	if unavail+readyPodCount >= newPodTarget {
 		// record the successful upgrade
+		klog.InfoS("pods are ready", "current batch", currentBatch)
+		c.recorder.Event(c.parentController, event.Normal("Batch Available",
+			fmt.Sprintf("the batch num = %d is available", c.rolloutStatus.CurrentBatch)))
 		c.rolloutStatus.StateTransition(v1alpha1.OneBatchAvailableEvent)
 	} else {
 		// continue to verify
+		klog.V(common.LogDebug).InfoS("the batch is not ready yet", "current batch", currentBatch)
 		c.rolloutStatus.StateTransition(v1alpha1.BatchRolloutVerifyingEvent)
 	}
 	return c.rolloutStatus
 }
 
-// Finalize makes sure the Cloneset is all upgraded and
+// FinalizeOneBatch makes sure that the rollout status are updated correctly
+func (c *CloneSetController) FinalizeOneBatch(ctx context.Context) *v1alpha1.RolloutStatus {
+	// nothing to do for now
+	return c.rolloutStatus
+}
+
+// Finalize makes sure the Cloneset is all upgraded
 func (c *CloneSetController) Finalize(ctx context.Context) *v1alpha1.RolloutStatus {
 	if c.fetchCloneSet(ctx) != nil {
 		return c.rolloutStatus
 	}
-	// mark the rollout finalized
-	c.recorder.Event(c.parentController, event.Normal("Finalized", "Rollout resource are finalized"))
+
 	c.rolloutStatus.StateTransition(v1alpha1.RollingFinalizedEvent)
+
 	return c.rolloutStatus
 }
 
-// The functions below are helper functions
+/* --------------------
+The functions below are helper functions
```

Comment:
why not use another file?

Reply:
they are still methods (I've already separated out the utility functions). We could separate out all the helper functions if this file is too large in the end (it's almost complete now so it's not too big yet)